lib/jellyfish/room.ex

```diff
@@ -147,7 +148,10 @@ defmodule Jellyfish.Room do
 
   @impl true
   def handle_call(:get_state, _from, state) do
-    {:reply, state, state}
+    all_tracks = Engine.get_tracks(state.engine_pid)
```

Comment:
That's because the `TrackAdded` event doesn't have track metadata inside. So each time track is added, its metadata is empty. 
There is no message as `TrackMetadataUpdated` so the only way to get metadata is by fetching it with the `get_tracks` function.
Since there is no way to have up-to-date info about metadata we're never putting it inside the Room's state.
Instead, when the state is requested we get all the info about tracks and return it.

Reply:
I think that it would be better to add a message like `TrackMetadataUpdated` instead of this because we could easily overload the engine, e.g, each request listing all available rooms besides asking for room process will also make a call to the engine instance.

Reply:
I have some doubts, about whether this can actually flood the engine with requests. But I can add such notification to the Engine.